### ISSUE TITLE
Tagging a new release for building shared libraries

### ISSUE THREAD

**Issue by ViralBShah** [MersenneTwister-Lab/dSFMT#5]
It would be nice to have a new release from master that has the `-DDSFMT_SHLIB` flag for building shared libraries.

I have been thinking about Makefile additions that would make it easy to build a shared library, but it is unclear how to do it with a simple Makefile that we have. Something like this would work - but is a bit cumbersome.

```
SHLIBFLAGS = -DDSFMT_SHLIB -fPIC -shared -DDSFMT_DO_NOT_USE_OLD_NAMES

libdsfmt.so: dSFMT.c dSFMT.h
        $(CC) $(CCFLAGS) $(SSE2FLAGS) $(SHLIBFLAGS) -DDSFMT_MEXP=19937 -o $@ dSFMT.c

libdsfmt.dylib: dSFMT.c dSFMT.h
	$(CC) $(CCFLAGS) $(SSE2FLAGS) $(SHLIBFLAGS) -DDSFMT_MEXP=19937 -o $@ dSFMT.c

libdsfmt.dll: dSFMT.c dSFMT.h
        $(CC) $(CCFLAGS) $(SSE2FLAGS) $(SHLIBFLAGS) -DDSFMT_MEXP=19937 -o $@ dSFMT.c
```

Perhaps the best thing to do is just document the availability of the `-DDSFMT_SHLIB` option. It is straightforward to then build the shared library on whichever platform you are. This is how we do it in Julia for all the OSes:
 https://github.com/JuliaPackaging/Yggdrasil/blob/fddff44e98287fc7ce76118b95ffa4dadcfbc6a6/D/dSFMT/build_tarballs.jl#L31

**Comment by ViralBShah** [comment]
I've opened #6 that updates the documentation to mention the flag. I think once that is merged, it would be great to have a new release.

**Comment by ViralBShah** [comment]
Hello @MSaito Just checking in to see if a version bump is possible. Thank you.

**Comment by ViralBShah** [comment]
Just checking if a new version can be tagged.

**Comment by majiang** [comment]
Thanks @ViralBShah for reminding!
Created v2.2.5 from master.
